test/metabase/util/debugging_test.clj

```diff
@@ -0,0 +1,42 @@
+(ns metabase.util.debugging-test
+  (:require  [clojure.test :refer :all]
+             [metabase.config :as config]
+))
+
+(def ^:private dbg-prn-used? (atom false))
+
+(defn- dbg-prn*
+  [p x]
+  (p x)
+  x)
+
+(defmacro dbg-prn
+  "`prn`s the argument and also returns it, which is useful for debugging. There is also a CI check to ensure no usages
+  of it sneak into master. With two arguments, uses the first as a printing function instead of `prn`."
+  ([x]
+   `(dbg-prn prn ~x))
```

Comment:
Clojure trivia question: without the backquote the macro worked but tests failed; with the backquote tests passed. Why? I legit have no idea.

Reply:
my guess is it would print at expand time rather than emitting a form which will print at eval time. so it prints to stdout before your with-out-string binding is in effect. there's also likely some issues with returning objects from the macroexpansion rather than forms.

Reply:
That ~makes sense, thanks